src/utils.ts

```diff
@@ -242,6 +262,69 @@ export const parseTx = (tx: RawTx): FullTx => {
   return parsedTx;
 };
 
+/**
+ * Syncs the latest mempool
+ *
+ * @generator
+ * @yields {MempoolEvent}
+ */
+export async function* syncLatestMempool(): AsyncGenerator<MempoolEvent> {
+
+  logger.info(`Downloading mempool...`);
+  let mempoolResp;
+  try {
+    mempoolResp = await downloadMempool();
+  } catch (e) {
+    yield {
+      success: false,
+      type: 'error',
+      message: 'Could not download from mempool api',
+    };
+    return;
+  }
+
+  for (let i = 0; i < mempoolResp.transactions.length; i++) {
+    const tx = await downloadTxFromId(mempoolResp.transactions[i]);
+
+    if (tx === null) {
+      yield {
+        type: 'wait',
+        success: false,
+        message: `Failure on transaction ${mempoolResp.transactions[i]} in mempool`,
+      };
+      return;
+    }
+
+    const preparedTx: PreparedTx = prepareTx(tx);
+
+    try {
+      const sendTxResponse: ApiResponse = await sendTx(preparedTx);
+      if (!sendTxResponse.success) {
+        logger.error(sendTxResponse);
+        throw new Error(sendTxResponse.message);
+      }
+    } catch (e) {
+      yield {
+        type: 'wait',
+        success: false,
+        message: `Failure on transaction ${preparedTx.tx_id} in mempool`,
+      };
+      return;
+    }
+
+    yield {
+      type: 'tx_success',
+      success: true,
+      txId: preparedTx.tx_id,
+    };
+  }
+
+  yield {
+    success: true,
+    type: 'finished',
+  };
+}
+
 /**
  * Syncs to the latest block
  *
```

Comment:
_Unrelated to line number:_

Update the gist on README with the new machine schema, you can do that by copying the code (and removing/mocking imports) and pasting to https://xstate.js.org/viz/, logging in with github and saving the gist

Reply:
Done.